win32ss/gdi/gdi32/objects/font.c

```diff
@@ -2062,32 +2062,44 @@ RemoveFontResourceA(LPCSTR lpFileName)
 }
 
 /*
- * @unimplemented
+ * @implemented
  */
 BOOL
 WINAPI
 RemoveFontResourceExA(LPCSTR lpFileName,
                       DWORD fl,
-                      PVOID pdv
-                     )
+                      PVOID pdv)
 {
     NTSTATUS Status;
     LPWSTR lpFileNameW;
+    BOOL result;
 
-    /* FIXME the flags */
-    /* FIXME the pdv */
-    /* FIXME NtGdiRemoveFontResource handle flags and pdv */
+    if (fl & ~(FR_PRIVATE | FR_NOT_ENUM))
+    {
+        SetLastError(ERROR_INVALID_PARAMETER);
+        return FALSE;
+    }
 
-    Status = HEAP_strdupA2W ( &lpFileNameW, lpFileName );
-    if (!NT_SUCCESS (Status))
-        SetLastError (RtlNtStatusToDosError(Status));
-    else
+    _SEH2_TRY
+    {
+        Status = HEAP_strdupA2W(&lpFileNameW, lpFileName);
+    }
+    _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
     {
+        DPRINT1("Exception!\n");
+        _SEH2_YIELD(return FALSE);
+    }
+    _SEH2_END;
 
-        HEAP_free ( lpFileNameW );
+    if (!NT_SUCCESS(Status))
```

Comment:
Side-remark (unrelated to this PR):
[HEAP_strdupA2W](https://git.reactos.org/?p=reactos.git;a=blob;f=win32ss/gdi/gdi32/misc/heap.c;hb=c35bb8dd733ac45dc6293fce6d9d3eabaca546af#l42) has an inconvenience/bug in which, if the actual heap allocation succeeded BUT the `RtlMultiByteToUnicodeN` call failed, the returned Status code would be a failure, however the allocated memory stays allocated.
Then, in cases one checks for failure like done here in this PR, one would leak the allocated buffer.
The proper fix would be in `HEAP_strdupA2W` to check whether `RtlMultiByteToUnicodeN` failed, and if so, free the allocated buffer and return the Status code.